tests/unit/test_backends.py

```diff
@@ -208,6 +213,21 @@ def testRunListReferenceBases(self):
         id_ = "example_1:simple"
         self.runListReferenceBases(id_)
 
+    def testOneDatasetRestriction(self):
+        # no datasetIds attr
+        request = protocol.SearchReadsRequest()
+        self._backend._getDatasetFromRequest(request)
+
+        # datasetIds attr
+        request = protocol.SearchVariantSetsRequest()
+        with self.assertRaises(exceptions.NotExactlyOneDatasetException):
+            self._backend._getDatasetFromRequest(request)
+        request.datasetIds = ['dataset1']
+        self._backend._getDatasetFromRequest(request)
```

Comment:
Maybe check that this is the expected value? Or perhaps we should have another separate test for this?